examples/hybrid/plane/inertial_gravity_wave.jl

```diff
@@ -59,42 +62,69 @@ jacobian_flags = (;
 )
 show_progress_bar = true
 
-if is_discrete_hydrostatic_balance
+function discrete_hydrostatic_balance!(ᶠΔz, ᶜΔz, grav)
+
     # Yₜ.f.w = 0 in implicit tendency                                        ==>
     # -(ᶠgradᵥ(ᶜp) / ᶠinterp(ᶜρ) + ᶠgradᵥᶜΦ) = 0                             ==>
     # ᶠgradᵥ(ᶜp) = -grav * ᶠinterp(ᶜρ)                                       ==>
     # (p(z + Δz) - p(z)) / Δz = -grav * (ρ(z + Δz) + ρ(z)) / 2               ==>
     # p(z + Δz) + grav * Δz * ρ(z + Δz) / 2 = p(z) - grav * Δz * ρ(z) / 2    ==>
     # p(z + Δz) * (1 + δ * Δz / 2) = p(z) * (1 - δ * Δz / 2)                 ==>
     # p(z + Δz) / p(z) = (1 - δ * Δz / 2) / (1 + δ * Δz / 2)                 ==>
-    # p(z) = p(0) * ((1 - δ * Δz / 2) / (1 + δ * Δz / 2))^(z / Δz)
-    p₀(z) = p_0 * ((1 - δ * Δz / 2) / (1 + δ * Δz / 2))^(z / Δz)
-else
-    p₀(z) = p_0 * exp(-δ * z)
+    # p(z + Δz) = p(z) * (1 - δ * Δz / 2) / (1 + δ * Δz / 2)
+    ᶜp = similar(ᶜΔz)
+    ᶜp1 = Fields.level(ᶜp, 1)
+    ᶜΔz1 = Fields.level(ᶜΔz, 1)
+    @. ᶜp1 = p_0 * (1 - δ * ᶜΔz1 / 4) / (1 + δ * ᶜΔz1 / 4)
+    for i in 1:(Spaces.nlevels(axes(ᶜp)) - 1)
+        ᶜpi = parent(Fields.level(ᶜp, i))
+        ᶜpi1 = parent(Fields.level(ᶜp, i + 1))
+        ᶠΔzi1 = parent(Fields.level(ᶠΔz, Spaces.PlusHalf(i)))
+        @. ᶜpi1 = ᶜpi * (1 - δ * ᶠΔzi1 / 2) / (1 + δ * ᶠΔzi1 / 2)
+    end
+    return ᶜp
```

Comment:
This block is the key feature in this PR for calculation of pressure assuming varying \delta z levels; broadcasts are then required in the initial condition functions, these appear to be done consistently (local build passes with comparable plots to the test case in main). Could you please include the plots generated from this new test-case in the buildkite-artifacts? (With this plot item resolved I can approve the PR)

Reply:
Hi Akshay, thank you so much for your thorough review, and yes, you got the main point of this PR excellently! 
And yes, you are right. I left the uniform grid as default, but I'll make sure this block of code is excercised in CI as well

Reply:
Is this a feature that we plan to support? If so, let's put this in `src/` instead to avoid code duplication.

Reply:
It's only specific to this example